docs/design/binary-platform-filtering.md

```diff
@@ -0,0 +1,279 @@
+
+- Optional platform fields are encapsulated within a `binary` submodel for relevant package types. This approach provides a clear namespace for binary-related options and allows for future extensions without cluttering the top-level package input.
+- Platform fields are handled at the PackageInput level rather than globally because the fields needed to filter/select prefetched binaries may differ between package managers
+- The `allow_binary` option will be renamed (deprecated) and moved to the `binary` submodel, but will be retained until the next major version for backwards compatibility
+- Platform fields will accept a list to allow the flexibility of multiple filters
+- Platform fields can use Literals for input validation for small, stable sets like Python implementations
+- Platform fields may have sensible defaults for the most common use cases. The existing `allow_binary` behavior must be maintained until that field is removed.
+- Introduce an optional packages field within the binary model. This field accepts a list of package names, acting as an allow-list for binary prefetching. If this list is provided, only packages named in it will be considered for binary downloads; all others will be treated as source-only. This option applies only to backends like pip and bundler where binary fetching is optional.
+
+```python
+class BinaryModeOptions(pydantic.BaseModel):
+    enabled: bool = False  # Renamed from allow_binary
+    packages: Optional[list[str]] = None  # List of packages to allow binaries for
+
+class PipBinaryOptions(BinaryModeOptions):
+    arch: Optional[list[str]] = ["x86_64"]
+    os: Optional[list[str]] = ["linux"]
+    py_version: Optional[list[str]] = None  # e.g., ["39", "310", "311"] (no dots)
+    py_impl: Optional[list[str]] = ["py", "cp"]  # e.g., ["cp", "pp", "py"]
+
+class BundlerBinaryOptions(BinaryModeOptions):
+    platform: Optional[list[str]] = None
+
+class RpmBinaryOptions(pydantic.BaseModel):
+    # binary.enabled is not relevant for RPM
+    arch: Optional[list[str]] = None
+
+class PipPackageInput(_PackageInputBase):
+    type: Literal["pip"]
+    allow_binary: bool = False  # Deprecated by binary.enabled, but kept for compatibility
+    binary: Optional[PipBinaryOptions] = None
+
+    @pydantic.model_validator(mode="after")
+    def _handle_legacy_allow_binary(self):
+        # If allow_binary is True and binary options are not set,
+        # initialize binary options with enabled=True.
+        # Issue a deprecation warning.
+
+class BundlerPackageInput(_PackageInputBase):
+    type: Literal["bundler"]
+    allow_binary: bool = False  # Deprecated by binary.enabled, but kept for compatibility
+    binary: Optional[BundlerBinaryOptions] = None
+
+    @pydantic.model_validator(mode="after")
+    def _handle_legacy_allow_binary(self):
+        ...
+
+class RpmPackageInput(_PackageInputBase):
+    type: Literal["rpm"]
+    binary: Optional[RpmBinaryOptions] = None
+```
+
+#### CLI Usage
```

Comment:
nitpick: this section splits the mental flow a bit IMO since the next section defines the strategies. It might be better to swap the order of the two, nonblocking though, feel free to ignore.